### PR TITLE
[FIX] deprecated QListWidget::setItemSelected()

### LINKED ISSUE #5942: warning: `QListWidget::setItemSelected()` is deprecated
```shell
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ListEditor.cpp: In member function ‘void OpenMS::Internal::ListTable::createNewRow()’:
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ListEditor.cpp:334:22: warning: ‘void QListWidget::setItemSelected(const QListWidgetItem*, bool)’ is deprecated: Use QListWidgetItem::setSelected() instead [-Wdeprecated-declarations]
  334 |       setItemSelected(item, true);
      |       ~~~~~~~~~~~~~~~^~~~~~~~~~~~
In file included from /usr/include/qt/QtWidgets/QListWidget:1,
                 from /home/USER/Development/OpenMs/src/openms_gui/include/OpenMS/VISUAL/ListEditor.h:45,
                 from /home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ListEditor.cpp:35:
/usr/include/qt/QtWidgets/qlistwidget.h:263:10: note: declared here
  263 |     void setItemSelected(const QListWidgetItem *item, bool select);
      |          ^~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by Amit0617**
# Description

Please include a summary of the change and which issue is fixed.
Fixes #5942 

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

# Advanced commands (admins / reviewer only):
- `/rebase` will try to rebase the PR on the current develop branch.
- `/reformat` (experimental) applies the clang-format style changes as additional commit
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Can you check that all the changes you want to make are available in Qt5.6 (the minimum required version).
If not, you can maybe just ignore the warnings with a pragma and a  comment.

**Comment by jpfeuffer** [comment]
Also this would never work, since you also need to change the header file:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms_gui/include/OpenMS/VISUAL/ListEditor.h#L112

Did you test this locally on your machine? Please check at least this before submitting a PR.
If you do not want to set up on your machine, you can use gitpod.io

File: src/openms_gui/source/VISUAL/ListEditor.cpp [hunk 1]
```diff
@@ -276,7 +277,7 @@ namespace OpenMS
     //LISTTABLE
     //////////////////////////////////////////////////////////////
     ListTable::ListTable(QWidget * parent) :
-      QListWidget(parent)
+      QListWidgetItem(parent)
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/ListEditor.cpp:280]: you are changing here the type from QListWidget to one of its items.

**Comment by Amit0617** [comment]
ahh... looks like commit history got really worse here if this doesn't looks right. I will create another clean pull

**Comment by Amit0617** [comment]
closing in favour of #5969

**Pull request closed without merging**